admin_guide/backup_restore.adoc

```diff
@@ -0,0 +1,224 @@
+xref:../install_config/install/advanced_install.adoc#install-config-install-advanced-install[Advanced
+Installation] method)
+- *_/etc/yum.repos.d/ose.repo_* (from the
+xref:../install_config/install/disconnected_install.adoc#install-config-install-disconnected-install[Disconnected
+Installation] method)
+
+. Install packages that provide various utility commands:
++
+----
+# yum install etcd
+----
+
+Note the location of the *etcd* data directory (or `$ETCD_DATA_DIR` in the
+following sections), which depends on how *etcd* is deployed.
+
+[options="header",cols="1,2"]
+|===
+| Deployment | Data Directory
+
+|all-in-one VM
+|*_/var/lib/openshift/openshift.local.etcd_*
+
+|external (not on master)
+|*_/var/lib/etcd_*
+
+|embedded (on master)
+|*_/var/lib/origin/etcd_*
+|===
+
+
+[[cluster-backup]]
+== Cluster Backup
+
+. Save all the certificates and keys, on each master:
++
+----
+# cd /etc/origin/master
+# tar cf /tmp/certs-and-keys-$(hostname).tar \
+    master.proxy-client.crt \
+    master.proxy-client.key \
+    proxyca.crt \
+    proxyca.key \
+    master.server.crt \
+    master.server.key \
+    ca.crt \
+    ca.key \
+    master.etcd-client.crt \
+    master.etcd-client.key \
+    master.etcd-ca.crt
+----
+
+. If *etcd* is running on more than one host, stop it on each host:
++
+----
+# sudo systemctl stop etcd
+----
++
+Although this step is not strictly necessary, doing so ensures that the *etcd*
+data is fully synchronized.
+
+. Create an *etcd* backup:
++
+----
+# etcdctl backup \
+    --data-dir $ETCD_DATA_DIR \
+    --backup-dir $ETCD_DATA_DIR.bak
+----
++
+[NOTE]
+====
+If *etcd* is running on more than one host,
+the various instances regularly synchronize their data,
+so creating a backup for one of them is sufficient.
+====
+
+. Create a template for all cluster API objects:
++
+====
+----
+$ oc export all \
+    --exact \//<1>
+    --all-namespaces \
+    --as-template=mycluster \//<2>
+    > mycluster.template.yaml
+----
+<1> Preserve fields that may be cluster specific,
+such as service `portalIP` values or generated names.
+<2> The output file has `kind: Template` and `metadata.name: mycluster`.
+====
++
+[IMPORTANT]
+====
+The object types included in `oc export all` are:
+
+----
+y	BuildConfig
+y	Build
+no-?	componentstatuses (aka 'cs')
+no-4	configmaps
+no-?	daemonsets (aka 'ds')
+y	DeploymentConfig
+no-?	deployments
+no-4	events (aka 'ev')
+no-4	endpoints (aka 'ep')
+no-2	horizontalpodautoscalers (aka 'hpa')
+no-1	imagestreamimages (aka 'isimage')
+y	ImageStream
+y	ImageStreamTag
+no-?	ingress (aka 'ing')
+no-2	groups
+no-?	jobs
+no-2	limitranges (aka 'limits')
+no-?	nodes (aka 'no')
+no-1	namespaces (aka 'ns')
+y	Pod
+no-?	persistentvolumes (aka 'pv')
+no-3	persistentvolumeclaims (aka 'pvc')
+no-2	policies
+no-1	projects
+no-2	quota
+no-2	resourcequotas (aka 'quota')
+no-?	replicasets (aka 'rs')
+y	ReplicationController
+no-2	rolebindings
+y	Route
+no-3	secrets
+no-2	serviceaccounts
+y	Service
+no-2	users
+----
```

Comment:
> @deads2k any idea on the formatting here? Thanks!

I don't recognize it the format, but the list looks correct (for now).  Seem like we might want to just define `all` and indicate that everything not listed isn't included.